### DISCUSSION TITLE
Unable to load any of {libcudnn_ops.so.9.1.0, libcudnn_ops.so.9.1, libcudnn_ops.so.9, libcudnn_ops.so}

### DISCUSSION THREAD

**Discussion by toanhuynhnguyen** [SYSTRAN/faster-whisper#1114] [Q&A] [answered]
After installing:
```
pip install --force-reinstall "faster-whisper @ https://github.com/SYSTRAN/faster-whisper/archive/refs/heads/master.tar.gz"
```

I run the code:
```
from faster_whisper import WhisperModel, BatchedInferencePipeline

model = WhisperModel("medium", device="cuda", compute_type="float16")
batched_model = BatchedInferencePipeline(model=model)
segments, info = batched_model.transcribe("audio.mp3", batch_size=16)

for segment in segments:
    print("[%.2fs -> %.2fs] %s" % (segment.start, segment.end, segment.text))
```

I get this error:
```
Unable to load any of {libcudnn_ops.so.9.1.0, libcudnn_ops.so.9.1, libcudnn_ops.so.9, libcudnn_ops.so}
Invalid handle. Cannot load symbol cudnnCreateTensorDescriptor
Aborted (core dumped)
```

Can anyone help me on this error, thank you so much.

**Comment by jhj0517** [comment, marked as answer]
Currently there's a strict version matrix between `faster-whisper` and `torch` : https://github.com/SYSTRAN/faster-whisper/issues/1086#issue-2612158224

You have two options to fix this,

1) downgrade `ctranslate2`
```
pip install faster-whisper ctranslate2==4.4.0
```
2) Or install `CUDA 12.4` and the latest `torch` 
```
pip install faster-whisper torch --extra-index-url https://download.pytorch.org/whl/cu124
```

**Reply by toanhuynhnguyen** [reply]
Tried with: pip install faster-whisper torch --index-url https://download.pytorch.org/whl/cu124
I got the below error:
Looking in indexes: https://download.pytorch.org/whl/cu124
ERROR: Could not find a version that satisfies the requirement faster-whisper (from versions: none)
ERROR: No matching distribution found for faster-whisper

**Reply by jhj0517** [reply]
@toanhuynhnguyen Sorry, try with `--extra-index-url` :

```
pip install faster-whisper torch --extra-index-url https://download.pytorch.org/whl/cu124
```

**Comment by sky-cake** [comment]
I install `cudnn9-cuda-12` through the instructions found at,
https://developer.nvidia.com/cudnn-downloads?target_os=Linux&target_arch=x86_64&Distribution=Debian&target_version=12&target_type=deb_local&Configuration=Full
